File: src/main/java/com/braydenoneal/blang/parser/statement/ContinueStatement.java
package com.braydenoneal.blang.parser.statement;

import com.braydenoneal.blang.parser.ParseException;
import com.braydenoneal.blang.parser.Program;
import com.braydenoneal.blang.tokenizer.Type;
import com.mojang.serialization.Codec;
import com.mojang.serialization.MapCodec;

public record ContinueStatement() implements Statement {
    @Override
    public Statement execute(Program program) {
        return this;
    }

    public static Statement parse(Program program) throws ParseException {
        program.expect(Type.KEYWORD, "continue");
        program.expect(Type.SEMICOLON);
        return new ContinueStatement();
    }

    public static final MapCodec<ContinueStatement> CODEC = Codec.unit(new ContinueStatement()).fieldOf("continue_statement");

    @Override
    public StatementType<?> getType() {
        return StatementTypes.CONTINUE_STATEMENT;
    }
}

File: src/main/java/com/braydenoneal/blang/parser/statement/ContinueStatement.kt
package com.braydenoneal.blang.parser.statement

import com.braydenoneal.blang.parser.Program
import com.braydenoneal.blang.tokenizer.Type
import com.mojang.serialization.Codec
import com.mojang.serialization.MapCodec

class ContinueStatement : Statement {
    override fun execute(program: Program): Statement {
        return this
    }

    override val type: StatementType<*> get() = StatementTypes.CONTINUE_STATEMENT

    companion object {
        fun parse(program: Program): Statement {
            program.expect(Type.KEYWORD, "continue")
            program.expect(Type.SEMICOLON)
            return ContinueStatement()
        }

        val CODEC: MapCodec<ContinueStatement> = Codec.unit(ContinueStatement()).fieldOf("continue_statement")
    }
}
